### PR TITLE
Added website versioning steps to the release procedures

### PR DESCRIPTION
**Pull request by sarahmaddox**
In the release procedures, split out the steps for updating version numbers in master website and creating a new website branch. Added a number of steps to the section on updating version numbers in various files

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4489)

### CODE REVIEW

File: docs_dev/releasing.md [hunk 1]
```diff
@@ -160,14 +160,92 @@ Alternatively you can use the UI.
...
+
+We usually start updating the website for the upcoming major/minor release when
+we've released a good RC for that release. At this point, it's useful for
+users to start experimenting with the upcoming release.
+
+When we're ready to start updating the docs for the upcoming version, we
+need to update various version numbers on the website, to ensure users are
+getting the correct installation instructions etc. Here are the steps to follow:
+
+1. Make sure the website branch for the current major/minor version has been
```
Comment (jlewi) [inline docs_dev/releasing.md]: Its somewhat ambiguous which version current is referring to; i.e. is it the version already released or the version we are releasing.

My suggestion would be in the preamble to say something like suppose we are in the process of releasing 0.Y so the docs are currently serving version 0.X. 

Then you can refer to 0.Y and 0.X below and its clear what you are refering to.

Comment (jlewi) [reply docs_dev/releasing.md]: Or you can just say "if we are releasing 0.7 then make sure the branch for v0.6 exists" that would be consistent with what you have below.

Comment (sarahmaddox) [reply docs_dev/releasing.md]: Good point. Done.

**Pull request merged**